google/cloud/aiplatform/datasets/tabular_datasets.py

```diff
@@ -0,0 +1,135 @@
+        display_name: str,
+        gcs_source: Optional[Sequence[str]] = None,
+        bq_source: Optional[str] = None,
+        labels: Optional[Dict] = None,
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        credentials: Optional[auth_credentials.Credentials] = None,
+    ) -> "TabularDataset":
+        """Creates a new tabular dataset.
+
+        Args:
+            display_name (str):
+                Required. The user-defined name of the Dataset.
+                The name can be up to 128 characters long and can be consist
+                of any UTF-8 characters.
+            gcs_source: Optional[Sequence[str]]=None:
+                Google Cloud Storage URI(-s) to the
+                input file(s). May contain wildcards. For more
+                information on wildcards, see
+                https://cloud.google.com/storage/docs/gsutil/addlhelp/WildcardNames.
+            bq_source: Optional[str]=None:
+                BigQuery URI to the input table.
+            labels: (Optional[Dict]) = None
+                The labels with user-defined metadata to organize your
+                Datasets.
+
+                Label keys and values can be no longer than 64 characters
+                (Unicode codepoints), can only contain lowercase letters,
+                numeric characters, underscores and dashes. International
+                characters are allowed. No more than 64 user labels can be
+                associated with one Dataset (System labels are excluded).
+
+                See https://goo.gl/xmQnxf for more information and examples
+                of labels. System reserved label keys are prefixed with
+                "aiplatform.googleapis.com/" and are immutable.
+            project: Optional[str]=None,
+                Project to upload this model to. Overrides project set in
+                aiplatform.init.
+            location: Optional[str]=None,
+                Location to upload this model to. Overrides location set in
+                aiplatform.init.
+            credentials: Optional[auth_credentials.Credentials]=None,
+                Custom credentials to use to upload this model. Overrides
+                credentials set in aiplatform.init.
+        Returns:
+            tabular_dataset (TabularDataset):
+                Instantiated representation of the managed tabular dataset resource.
+        """
+
+        metadata_schema_uri = schema.dataset.metadata.tabular
+
+        dataset_obj = super().create(
+            display_name=display_name,
+            metadata_schema_uri=metadata_schema_uri,
+            gcs_source=gcs_source,
+            bq_source=bq_source,
+            labels=labels,
+            project=project,
+            location=location,
+            credentials=credentials,
+        )
+
+        return dataset_obj
+
+    @property
+    def import_data(self):
+        raise AttributeError("'TabularDataset' object has no 'import_data' method")
```

Comment:
I wonder if it's better to remove the function and make it private in the superclass. Only implement it for subclasses where importing makes sense.

Reply:
As discussed with @morgandu, I think we could later refactor Datasets.py to remove the import_data functionality from it and move it to an DatasetImportable abstract class.

Fundamentally, not ALL Datasets have import functionality (e.g.. tabular).

TabularDataset would subclass Dataset AND DatasetImportable.
Vision, NLP and others would just subclass Dataset.

Then you won't have this weird import_data function in a class that doesn't need it. And you've decoupled import functionality from core Dataset functionality.

Reply:
I agree that this issue should be resolved in the class hierarchy.

I think though that we still want a generic Dataset class (with name Dataset) similar to what we have now that supports all methods. 

> TabularDataset would just subclass Dataset.
Vision, NLP and others would subclass Dataset AND DatasetImportable.

If we go the multiple inheritance route then probably something like this:
TabularDataset would just subclass _Dataset.
Dataset, Vision, NLP and others would subclass _Dataset AND _DatasetImportable

But we can also do this without mixin:
TabularDataset would subclass _Dataset.
Dataset is a subclass of _Dataset with import and associated methods
Vision, NLP and others would subclass Dataset

To me it seems like it's cleaner without multiple inheritance because _DatasetImportable has a hard requirement on _Dataset to work.

Reply:
That makes sense, although we might run into the diamond problem if we need to compose more functionality later on. For example, if we later have _DatasetExportable and there's a subclass that implements both _DatasetImportable and _DatasetExportable. Then we have to worry about method resolution order and such. 

Just pointing that out but I'm fine either way as I believe breaking out the Import functionality is a step in the right direction in any case.

Reply:
Also the naming seems strange to me. Dataset seems like it should have LESS functionality than a class name ImportableDataset.

Reply:
Agree that this should be changed to `NotImplementedError`.